### PR TITLE
Include child scenes when uploading to Mozart

### PR DESCRIPTION
**Pull request by abarth**
Previously we just dropped child scenes on the floor. Now we upload them
to Mozart. However, we just draw them on top of all the other content
and don't apply any clips or blends.

### CODE REVIEW

**Comment by abarth** [comment]
@j9brown @chinmaygarde

File: flow/layers/child_scene_layer.cc [hunk 1]
```diff
@@ -4,15 +4,42 @@
...
+}
+
 void ChildSceneLayer::Paint(PaintContext::ScopedFrame& frame) {
 }
 
+void ChildSceneLayer::UpdateScene(mojo::gfx::composition::SceneUpdate* update,
+                                  mojo::gfx::composition::Node* container) {
+  uint32_t id = next_id++;
+
+  auto child_resource = mojo::gfx::composition::Resource::New();
```
Comment (Hixie) [inline flow/layers/child_scene_layer.cc]: what will this do on platforms that don't have mozart?

File: flow/layers/layer_tree.h [hunk 2]
```diff
@@ -22,6 +22,12 @@ class LayerTree {
 
   void Raster(PaintContext::ScopedFrame& frame);
 
+  // TODO(abarth): Integrate scnene updates with the rasterization pass so that
```
Comment (Hixie) [inline flow/layers/layer_tree.h]: nit: not that it matters, but typo "scene"

**Comment by Hixie** [comment]
lgtm

File: flow/layers/child_scene_layer.cc (hunk 1 above)
Comment (abarth) [inline flow/layers/child_scene_layer.cc]: This function is only called when we have mozart.

Comment (Hixie) [inline flow/layers/child_scene_layer.cc]: What stops it from being called when you don't have mozart?

Comment (abarth) [inline flow/layers/child_scene_layer.cc]: rasterizer_mojo.cc is the only caller and that's only linked into the binary in `flutter.mojo`.

Comment (Hixie) [inline flow/layers/child_scene_layer.cc]: Ah, I see. And Paint() is a no-op for this class, so in other rasterisers it just doesn't do anything.

This is relatively subtle. The API looks like it's just a normal part of the API, and the dart:ui side of this is always exposed, so someone could easily come along and try to call UpdateScene thinking it's broken that the Dart-exposed API is a no-op. Can you add a comment in layer.h next to the UpdateScene abstract method declaration explaining UpdateScene's role?

Comment (abarth) [inline flow/layers/child_scene_layer.cc]: Yeah, I'm not entirely sure how to handle the bifurcation of what works in standalone Flutter versus what works in Mojo-land.  I'm happy to add a comment.

**Pull request merged**